x-pack/plugins/security_solution/public/resolver/view/use_cube_assets.ts

```diff
@@ -0,0 +1,121 @@
+import { useMemo } from 'react';
+import { ResolverProcessType } from '../types';
+import { useUiSetting } from '../../../../../../src/plugins/kibana_react/public';
+import { useSymbolIDs } from './use_symbol_ids';
+import { useColors } from './use_colors';
+
+/**
+ * Provides colors and HTML IDs used to render the 'cube' graphic that accompanies nodes.
+ */
+export function useCubeAssets(
+  isProcessTerminated: boolean,
+  isProcessTrigger: boolean
+): NodeStyleConfig {
+  const SymbolIds = useSymbolIDs();
+  const isDarkMode = useUiSetting('theme:darkMode');
+  const theme = isDarkMode ? euiThemeAmsterdamDark : euiThemeAmsterdamLight;
+  const colorMap = useColors();
+
+  const nodeAssets: NodeStyleMap = useMemo(
+    () => ({
+      runningProcessCube: {
+        backingFill: colorMap.processBackingFill,
+        cubeSymbol: `#${SymbolIds.runningProcessCube}`,
+        descriptionFill: colorMap.descriptionText,
+        descriptionText: i18n.translate('xpack.securitySolution.endpoint.resolver.runningProcess', {
+          defaultMessage: 'Running Process',
+        }),
+        isLabelFilled: true,
+        labelButtonFill: 'primary',
+        strokeColor: theme.euiColorPrimary,
+      },
+      runningTriggerCube: {
+        backingFill: colorMap.triggerBackingFill,
+        cubeSymbol: `#${SymbolIds.runningTriggerCube}`,
+        descriptionFill: colorMap.descriptionText,
+        descriptionText: i18n.translate('xpack.securitySolution.endpoint.resolver.runningTrigger', {
+          defaultMessage: 'Running Trigger',
+        }),
+        isLabelFilled: true,
+        labelButtonFill: 'danger',
+        strokeColor: theme.euiColorDanger,
+      },
+      terminatedProcessCube: {
+        backingFill: colorMap.processBackingFill,
+        cubeSymbol: `#${SymbolIds.terminatedProcessCube}`,
+        descriptionFill: colorMap.descriptionText,
+        descriptionText: i18n.translate(
+          'xpack.securitySolution.endpoint.resolver.terminatedProcess',
+          {
+            defaultMessage: 'Terminated Process',
+          }
+        ),
+        isLabelFilled: false,
+        labelButtonFill: 'primary',
+        strokeColor: theme.euiColorPrimary,
+      },
+      terminatedTriggerCube: {
+        backingFill: colorMap.triggerBackingFill,
+        cubeSymbol: `#${SymbolIds.terminatedTriggerCube}`,
+        descriptionFill: colorMap.descriptionText,
+        descriptionText: i18n.translate(
+          'xpack.securitySolution.endpoint.resolver.terminatedTrigger',
+          {
+            defaultMessage: 'Terminated Trigger',
+          }
+        ),
+        isLabelFilled: false,
+        labelButtonFill: 'danger',
+        strokeColor: theme.euiColorDanger,
+      },
+    }),
+    [SymbolIds, colorMap, theme]
+  );
+
+  if (isProcessTerminated) {
+    if (isProcessTrigger) {
+      return nodeAssets.terminatedTriggerCube;
+    } else {
+      return nodeAssets[processTypeToCube.processTerminated];
```

Comment:
I'm not sure we need this indirection, but I'll leave that suggestion to a future PR